lib/strings.nix

```diff
@@ -1010,7 +1010,7 @@ rec {
        => "-nix-store-2g75chlbpxlrqn15zlby2dfh8hr9qwbk-hello-2.10"
   */
   sanitizeDerivationName =
-  let okRegex = match "[[:alnum:]+_?=-][[:alnum:]+._?=-]*";
+  let okRegex = match "[[:alnum:]+_=-][[:alnum:]+._=-]*";
```

Comment:
This regex is only used for an early return if the string is already a valid derivation name. By making it match less derivation names, the result is not affected:
```
nix-repl> lib.strings.sanitizeDerivationName "?"     
"?"
```

To actually make it work you need to also change a regex further down.

Reply:
Oh oops, this was a year-old commit, I may have neglected to test it, So that's embarrassing 😅

I'll mark this as draft and fix it in a bit